lib/Index/FilePathIndex.cpp

```diff
@@ -165,7 +165,9 @@ bool FileIndexImpl::foreachMainUnitContainingFile(CanonicalFilePathRef filePath,
       currUnit.UnitName = unitInfo.UnitName;
       currUnit.ModTime = unitInfo.ModTime;
       currUnit.MainFilePath = reader.getFullFilePathFromCode(unitInfo.MainFileCode);
-      currUnit.OutFilePath = reader.getFullFilePathFromCode(unitInfo.OutFileCode);
+      // Roundtrip is OK here since it assumes the path is already canonicalized so it won't
+      // change the out file path at all.
+      currUnit.OutFilePath = reader.getFullFilePathFromCode(unitInfo.OutFileCode).getPath();
```

Comment:
I'd prefer just adding another method that returns a string rather than one that returns a `CanonicalFilePathRef`, even with the comment.